flytekit/common/tasks/sdk_runnable.py

```diff
@@ -26,12 +26,13 @@ class ExecutionParameters(object):
     decorated function.
     """
 
-    def __init__(self, execution_date, tmp_dir, stats, execution_id, logging):
+    def __init__(self, execution_date, tmp_dir, stats, execution_id, logging, distributed_training_context=None):
```

Comment:
Not a huge fan of this but I'm okay with it if it's required.  Possible alternative... 
1. we can subclass EngineContext, and then we can just add any field we want
2. change the name to `task_specific_context` or something, but i’m struggling with how other tasks might use it and where/how tasks would set it.

Reply:
I like the subclass idea. I've made the change in the following commits